fbw-a32nx/src/systems/instruments/src/RMP/Components/NavRadioPanel.tsx

```diff
@@ -101,49 +101,71 @@ export const NavRadioPanel = (props: Props) => {
   let standbyWindow: JSX.Element;
 
   let index = props.side === 'L' ? 1 : 2;
-  if (props.transceiver === TransceiverType.ILS) {
+  if (props.receiver === TransceiverType.ILS) {
     index = 3; // Both RMPs manage the same ILS
   }
 
   const [mode, setMode] = useState(Mode.FREQUENCY);
 
-  const [activeFrequency, setActiveFrequencySaved] = useActiveFrequency(props.transceiver, props.side);
-  const [standbyFrequency, setStandbyFrequencySaved] = useStandbyFrequency(props.transceiver, props.side);
-  const [course, setCourseSaved] = useCourse(props.transceiver, props.side);
-
-  const [, setCourse] = useSimVar(
-    props.transceiver === TransceiverType.VOR ? `K:VOR${index}_SET` : 'L:A32NX_FM_LS_COURSE',
-    'number',
-    100,
-  );
+  const [activeFrequency, setActiveFrequencySaved] = useActiveFrequency(props.receiver, props.side);
+  const [standbyFrequency, setStandbyFrequencySaved] = useStandbyFrequency(props.receiver, props.side);
+  const [course, setCourseSaved] = useCourse(props.receiver, props.side);
 
   const [APPR] = useSimVar('L:A32NX_FCU_APPR_MODE_ACTIVE', 'bool');
   const [AP] = useSimVar('L:A32NX_AUTOPILOT_ACTIVE', 'bool');
   const [RA1] = useSimVar('L:A32NX_RA_1_RADIO_ALTITUDE', 'number');
   const [RA2] = useSimVar('L:A32NX_RA_2_RADIO_ALTITUDE', 'number');
+  const [, setCourse] = useSimVar(
+    props.receiver === TransceiverType.VOR ? `K:VOR${index}_SET` : 'L:A32NX_FM_LS_COURSE',
```

Comment:
The ACP should never write FM vars (or any other system's vars).

Reply:
Then I have to refactor it now the Rust code exists

Reply:
For the time being, it has to be done this way (or in Rust) since the FMGC is not complete